src/server/qgswmsserver.cpp

```diff
@@ -1446,10 +1434,16 @@ QImage* QgsWmsServer::getMap( HitTest* hitTest )
   applyOpacities( layersList, bkVectorRenderers, bkRasterRenderers, labelTransparencies, labelBufferTransparencies );
 
   if ( hitTest )
-    runHitTest( &thePainter, *hitTest );
+    runHitTest( mapSettings, &thePainter, *hitTest );
   else
   {
-    mMapRenderer->render( &thePainter );
+    QgsMapRendererCustomPainterJob renderJob( mapSettings, &thePainter );
+#ifdef HAVE_SERVER_PYTHON_PLUGINS
+    renderJob.setFeatureFilterProvider( mAccessControl );
+#endif
+    renderJob.renderSynchronously();
+    //renderJob.start();
+    //renderJob.waitForFinished();
   }
```

Comment:
A timeout is raised during unit tests when the start/waitForFinished functions are used. Do you know why?